### PR TITLE
AP_IntertialNav clarify 2D vs 3D uses and get_bearing_cd() takes Vector2f

### PR DESCRIPTION
**Pull request by hendjoshsr71**
The purpose of this PR is to make the following functions take Vector2f as that is all they use
`get_bearing_cd()`   & `get_horizontal_distance_cm()`

The other purpose is to simplify & clarify places where `inav.position()` is being used, but only in 2D to be `inav.position_xy()` My reasoning for this is I think it will help as an in-between step to remove NEU by making it clearer what is only XY vs Z in later PRs.

It also uses `inav.get_altitude()` or `inav.get_velocity_z()` where appropriate. 

Not sure how useful it is?
This PR saves 380+ bytes on Durandal for copter & just a wee bit elsewhere.

There are a couple spots with PosTypes where attention must be paid to where and how conversion is happening to keep it functionally equivalent.

### CODE REVIEW

**Comment by khancyr** [comment]
we should kill inertial_nav now that the inertial backend are better

File: libraries/AP_InertialNav/AP_InertialNav.cpp
Comment (tridge) [inline libraries/AP_InertialNav/AP_InertialNav.cpp:59]: add get_position_z_up_cm()

Comment (hendjoshsr71) [reply libraries/AP_InertialNav/AP_InertialNav.cpp:59]: Fixed

**Comment by hendjoshsr71** [comment]
Renamed to `inav.get_position_z_up_cm`

I also addressed some of the `zzz.length()` spots to use `length_squared()`

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
pending approval from @lthall

**Review by rmackay9 — APPROVED** [review]

**Review by lthall — APPROVED** [review]

File: ArduCopter/mode_follow.cpp
Comment (hendjoshsr71) [reply ArduCopter/mode_follow.cpp]: Outdated suggestion? Was fixed after approval.

File: ArduCopter/mode_follow.cpp
Comment (hendjoshsr71) [reply ArduCopter/mode_follow.cpp]: Outdated suggestion? Was fixed after approval.

File: libraries/AC_AttitudeControl/AC_PosControl.cpp
Comment (hendjoshsr71) [reply libraries/AC_AttitudeControl/AC_PosControl.cpp]: Outdated suggestion? Was fixed after approval.

**Comment by hendjoshsr71** [comment]
@lthall I think your comments were left over from the first DevCall of the week? Each of them had been addressed after your approval.

**Review by lthall — APPROVED** [review]

**Comment by rmackay9** [comment]
I've added the DevCallTopic but that's just so we don't forget about this but we can actually merge any time.

**Comment by rmackay9** [comment]
Sorry to say but this change makes back porting @lthall's recent fixes to 4.1 very difficult..

**Comment by hendjoshsr71** [comment]
Sorry about that. 

I'm not sure if there is a better way to move forward on getting rid ned/neu/cm?

**Pull request merged**
